### DISCUSSION TITLE
Restore all windows across a desktops and displays

### DISCUSSION THREAD

**Discussion by keithharper8** [rxhanson/RectanglePro-Community#573] [Q&A]
Apologies if this is here in the community somewhere, I spent awhile searching and found questions similar but not identical to what I am looking for. I need to be able to type a keyboard shortcut to reposition multiple apps across multiple desktops and displays like the app "Stay" from CordlessDog does. The app has worked for me for years and still works about half of the time but it has become unreliable and I have not seen updates in years. I have used Rectangle for years along with Stay and have been happy until recently. I looked around and was delighted to find that Rectangle Pro was recommended as a Stay replacement since Rectangle has been solid for me. I am maybe missing something, but I cannot figure out a way to restore all windows with one keyboard shortcut like I can with Stay. Currently I have to navigate to each display and desktop one at a time and type the same keyboard shortcut each time I switch. This works, but it sure would be incredible if it worked for all at once like Stay. Is this possible with rectangle Pro?

**Comment by rxhanson** [comment]
Thanks for checking out Rectangle Pro! 

Applying a layout to multiple displays at once is possible from the add layout button in App Layouts. Additionally, you can save a layout for each display and then drag and drop each of the window configurations into one layout. 

Applying layouts to spaces that are not visible is not possible because Apple never released a public API for this, unfortunately.

**Reply by dkniffin** [reply]
I'm also interested in this functionality. I did figure out how to do the multi-monitor configuration, but I'd LOVE to see one for multiple workspaces/desktops. It's a shame Apple didn't release a public API for it though :/
